### PR TITLE
{Misc.} Fix links and change master -> main

### PR DESCRIPTION
**Pull request by heaths**
Found a broken link (extra "b") and change instances of `master` branch to `main` (verified all were correct).

### CODE REVIEW

**Comment by yonzhan** [comment]
Good catch @heaths.

File: doc/extensions/authoring.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ Address comments as appropriate and consult the Azure CLI team if something is u
 
 ### Publish
 
-**For the extension whose source code is hosted in [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb)**, we will release for you once your code is merged into `master` branch. You must not update [index.json](https://github.com/Azure/azure-cli-extensions/blob/master/src/index.json) manually in this case.
+**For the extension whose source code is hosted in [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensions)**, we will release for you once your code is merged into `main` branch. You must not update [index.json](https://github.com/Azure/azure-cli-extensions/blob/main/src/index.json) manually in this case.
```
Comment (jiasli) [inline doc/extensions/authoring.md:44]: This was already fixed by https://github.com/Azure/azure-cli/pull/21851.

Your branch seems too old:

```
> git log --graph
* commit 2b96e6ffaa26c1dac24dda71a3096bcf65ad24e7 (HEAD -> patch-1, heaths/patch-1)
| Author: Heath Stewart <heaths@microsoft.com>
| Date:   Tue Aug 2 11:33:32 2022 -0700
|
|     Fix links and change master -> main
|
|     Found a broken link (extra "b") and change instances of `master` branch to `main` (verified all were correct).
|
* commit f70a427035f4577c817942968183d475a415bcb2 (tag: azure-cli-2.29.0, azure/master)
| Author: Jiashuo Li <4003950+jiasli@users.noreply.github.com>
| Date:   Fri Oct 8 17:00:24 2021 +0800
|
|     {Misc.} Add PR ID (#19809)
```

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
I have rebased your commit to latest `dev`.

File: doc/extensions/authoring.md (hunk 1 above)
Comment (heaths) [reply doc/extensions/authoring.md:44]: I clicked "Edit" in GitHub against your default branch. The docs from when I browsed here were what was currently visible. Is your default branch set correctly?

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**